packages/realm-server/tests/indexing-test.ts

```diff
@@ -236,19 +236,23 @@ module('indexing', function (hooks) {
   });
 
   test('can store card pre-rendered html in the index', async function (assert) {
-    let entry = await realm.searchIndex.searchEntry(
-      new URL(`${testRealm}mango`),
-    );
-    assert.strictEqual(
-      trimCardContainer(stripScopedCSSAttributes(entry!.isolatedHtml!)),
-      cleanWhiteSpace(`<h1> Mango </h1>`),
-      'pre-rendered html is correct',
-    );
+    let entry = await realm.searchIndex.instance(new URL(`${testRealm}mango`));
+    if (entry?.type === 'instance') {
+      assert.strictEqual(
+        trimCardContainer(stripScopedCSSAttributes(entry!.isolatedHtml!)),
+        cleanWhiteSpace(`<h1> Mango </h1>`),
+        'pre-rendered html is correct',
+      );
+    } else {
+      assert.ok(false, 'expected index entry not to be an error');
```

Comment:
I see this is already an established pattern (and exists above too) but is there a way to precise about this assertion vs using `assert.ok(false…`? Is the `if` check to prevent an exception?

Reply:
The `if` is for types. without the `if` we'd get typing errors trying to use the properties in the object we are trying to test